pkg/bgpv1/types/bgp.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"context"
+	"net"
+
+	"github.com/cilium/cilium/api/v1/models"
+	v2alpha1api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+)
+
+// BGPGlobal contains high level BGP configuration for given instance.
+type BGPGlobal struct {
+	ASN                   uint32
+	RouterID              string
+	ListenPort            int32 // When -1 gobgp won't listen on tcp:179
+	RouteSelectionOptions *RouteSelectionOptions
+}
+
+// RouteSelectionOptions contains generic BGP route selection tuning parameters
+type RouteSelectionOptions struct {
+	// AdvertiseInactiveRoutes when set will advertise route even if it is not present in RIB
+	AdvertiseInactiveRoutes bool
+}
+
+// Advertisement is a container object which associates a net.IPNet
+//
+// The `Net` field makes comparing this Advertisement with another IPNet encoded
+// prefixes simple.
+//
+// The `GoBGPPathUUID` field is a gobgp.GoBGPPathUUID object which can be forwarded to gobgp's
```

Comment:
Nit: gobgp.GoBGPPathUUID => gobgp.PathUUID?

Reply:
oh this is result of goland magic refactor :)